test/Test/Cardano/Prelude/GHC/Heap/Size.hs

```diff
@@ -0,0 +1,128 @@
+{-# LANGUAGE BangPatterns               #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MagicHash                  #-}
+{-# LANGUAGE OverloadedStrings          #-}
+{-# LANGUAGE TemplateHaskell            #-}
+
+{-# OPTIONS_GHC -fno-full-laziness #-}
+
+module Test.Cardano.Prelude.GHC.Heap.Size (tests) where
+
+import Cardano.Prelude hiding (diff)
+
+import GHC.Prim (Int#)
+import qualified Data.ByteString as BS.Strict
+
+import Hedgehog
+
+{-------------------------------------------------------------------------------
+  Various unboxed tuples, for testing purposes
+
+  We don't use an UNPACK pragma here because that is ignored at @-O0@, which
+  could result in false negatives.
+-------------------------------------------------------------------------------}
+
+data Ints2 = Ints2 Int# Int#
+data Ints3 = Ints3 Int# Int# Int#
+data Ints4 = Ints4 Int# Int# Int# Int#
+data Ints5 = Ints5 Int# Int# Int# Int# Int#
+
+{-------------------------------------------------------------------------------
+  Auxiliary
+-------------------------------------------------------------------------------}
+
+newtype NumWords = NumWords Int
+  deriving (Show, Eq, Ord, Num)
+
+-- | Size of a word (in bytes)
+wordSize :: Int
+wordSize = finiteBitSize (0 :: Word) `div` 8
```

Comment:
We could also use `wORD_SIZE` from `GHC.Exts.Heap.Constants`. It is defined as:

```
wORD_SIZE = #const SIZEOF_HSWORD
```

Reply:
Ah, great. Will use.